pvlib/clearsky.py

```diff
@@ -687,7 +687,7 @@ def detect_clearsky(measured, clearsky, times, window_length,
         raise NotImplementedError('algorithm does not yet support unequal ' \
                                   'times. consider resampling your data.')
 
-    samples_per_window = int(window_length / sample_interval)
+    samples_per_window = int(window_length / sample_interval) + 1
```

Comment:
Revert this change please.  `samples_per_window` is counting intervals, not endpoints. Because it's an intermediate it could be renamed to be more clear.

Reply:
I believe that not adding 1 gives the incorrect number of samples per window.  For example, if my data is 30-minute frequency (`sample_interval=30`) and I want 60 minute windows (`window_length=60`), the current implementation would only give 2 points per window (when the Hankel matrix is constructed in the following lines).  In this case, 2 points per window would only span a 30 minute window, not the intended 60.

Reply:
Your example makes my point. The algorithm operates on intervals not on points in time. The value at a timestamp is considered as the value for the following (?) interval - I'd have to look carefully at the Hankel matrix and the diff to see if we adopted a left- or right- endpoint convention.